src/cosmology/cosmology.cpp

```diff
@@ -42,8 +45,24 @@ void Cosmology::Initialize(struct Parameters *P, Grav3D &Grav, Particles3D &Part
   delta_a     = max_delta_a;
 
   // Initialize Time and set the time conversion
-  t_secs          = 0;
   time_conversion = KPC;
+  t_secs          = 0;
+
+  // get time offset
+  Real da_t_sec = 1.0e-2 * current_a;
+  // small but non-zero
+  Real a_t_sec = 1.0e-6;
+  Real dt_physical;
+  while (a_t_sec < current_a) {
+    if (a_t_sec + da_t_sec > current_a) {
+      da_t_sec = current_a - a_t_sec;
+    }
+    dt_physical = Get_dt_from_da(da_t_sec, a_t_sec);
+    t_secs += dt_physical * time_conversion;
+    a_t_sec += da_t_sec;
+    // chprintf(" Revised a_t_sec %f da_t_sec %f t_start : %e Myr\n", a_t_sec, da_t_sec, t_secs / MYR);
+  }
+  chprintf(" Revised t_start : %f Myr\n", t_secs / MYR);
```

Comment:
Would you please add a comment or two to this block of code clarifying the purpose?
- specifically, could you describe the purpose of this logic? It seems like the goal is to modify `t_secs`.
   - are we just setting ``t_start`` to the age of the universe at ``current_a``? (If so, why didn't we do this before?)
   - or are we doing something else? 
- Could you address what the while loop is specifically doing?

Reply:
I'll add a comment in the next commit.

I am indeed just modifying t_start to the age of the universe at current_a.  This is not so important for LCDM cosmologies, but when checking the expansion history for a non-standard cosmology it is very helpful.  Another issue is that the ICs will not always start at the same redshift, so `t_secs` is not measured relative to z~\infty.

The while loop advances the integration until the scale factor corresponding to `t_secs` is the same as the initial scale factor of the ICs (or restart) file.

Reply:
I've addressed this in commit 2b3b2a0f